azure/scope/cluster.go

```diff
@@ -230,18 +231,22 @@ func (s *ClusterScope) RouteTableSpecs() []azure.RouteTableSpec {
 }
 
 // NatGatewaySpecs returns the node nat gateway.
-func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
-	natGateways := []azure.NatGatewaySpec{}
+func (s *ClusterScope) NatGatewaySpecs() []azure.ResourceSpecGetter {
+	natGateways := []azure.ResourceSpecGetter{}
 
 	// We ignore the control plane nat gateway, as we will always use a LB to enable egress on the control plane.
+	// TODO: do we need to use setSubnet() in natgateways.go, since it doesn't add/remove any NodeSubnets() or change IsNatGatewayEnabled()
 	for _, subnet := range s.NodeSubnets() {
 		if subnet.IsNatGatewayEnabled() {
-			natGateways = append(natGateways, azure.NatGatewaySpec{
-				Name: subnet.NatGateway.Name,
+			natGateways = append(natGateways, &natgateways.NatGatewaySpec{
+				Name:           subnet.NatGateway.Name,
+				ResourceGroup:  s.ResourceGroup(),
+				SubscriptionID: s.SubscriptionID(),
+				Location:       s.Location(),
 				NatGatewayIP: infrav1.PublicIPSpec{
 					Name: subnet.NatGateway.NatGatewayIP.Name,
 				},
-				Subnet: subnet,
+				SubnetName: subnet.Name,
```

Comment:
if you're not using SubnetName anywhere after removing it from OwnerResourceName, this can probably be removed